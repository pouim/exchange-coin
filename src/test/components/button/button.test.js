import { shallow } from 'enzyme';
import Button from 'src/components/ui/button';
import { findByTestAttr } from 'src/test/testUtils';

describe('Button', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Button debug />);

        expect(component).toMatchSnapshot();
    });

    it('should render correctly with no props', () => {
        const component = shallow(<Button />);

        expect(component).toMatchSnapshot();
    });

    it('should render children correctly', () => {
        const testTitle = 'test';
        const component = shallow(<Button>{testTitle}</Button>);

        const element = findByTestAttr(component, 'custom-button');

        expect(element.text()).toBe('test');
    });

    it('should not render loading when it is false', () => {
        const component = shallow(<Button loading={false} />);
        const loadingSpin = component.find('[data-test="custom-button-loading"]');

        expect(loadingSpin).toHaveLength(0);
    });

    it('should not render loading when it is true', () => {
        const component = shallow(<Button loading={true} />);
        const loadingSpin = component.find('[data-test="custom-button-loading"]');

        expect(loadingSpin).toHaveLength(1);
    });
});
