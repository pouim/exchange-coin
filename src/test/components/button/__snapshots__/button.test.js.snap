// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Button should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly with no props 1`] = `ShallowWrapper {}`;
